File: Solutions/905. Sort Array By Parity/905. Sort Array By Parity.java
class Solution {
    public int[] sortArrayByParity(int[] A) {
        for(int i=0,j=0;j<A.length;++j){
            if(A[j]%2==0){
                swap(A,i,j);
                ++i;
            }
        }
        return A;
    }
    private void swap(int[] a,int i,int j){
        int tmp=a[i];
        a[i]=a[j];
        a[j]=tmp;
    }
}

File: Solutions/905. Sort Array By Parity/905. Sort Array By Parity.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& A) {
        for(int i=0,j=0;j<A.size();++j){
            if(A[j]%2==0){
                int tmp=A[i];
                A[i++]=A[j];
                A[j]=tmp;
            }
        }
        return A;
    }
};